src/ML/src/MANN.cpp

```diff
@@ -0,0 +1,385 @@
+                                   + numberOfJoints // joints positions
+                                   + numberOfJoints // joints velocities
+                                   + 3; // x y and omega component of the velocity
+
+    // check if output size is equal to the size of the mean and of the standard deviation
+    if (outputSize != m_pimpl->structuredOutput.mean.size()
+        || outputSize != m_pimpl->structuredOutput.std.size())
+    {
+        log()->error("{} The size of the mean and the standard deviation are different from the "
+                     "expected one. Mean size: {} Standard Deviation size: {} expected: {} ",
+                     logPrefix,
+                     m_pimpl->structuredOutput.mean.size(),
+                     m_pimpl->structuredOutput.std.size(),
+                     outputSize);
+        return false;
+    }
+
+    // resize the output
+    m_pimpl->structuredOutput.rawData.resize(outputSize);
+    m_pimpl->structuredOutput.shape[0] = 1; // batch
+    m_pimpl->structuredOutput.shape[1] = outputSize;
+
+    // create tensor required by onnx
+    m_pimpl->structuredOutput.tensor
+        = Ort::Value::CreateTensor<float>(m_pimpl->memoryInfo,
+                                          m_pimpl->structuredOutput.rawData.data(),
+                                          m_pimpl->structuredOutput.rawData.size(),
+                                          m_pimpl->structuredOutput.shape.data(),
+                                          m_pimpl->structuredOutput.shape.size());
+    // populate variable handler related to the output
+    // the serialization matters
+    m_pimpl->structuredOutput.handler.addVariable("future_base_positions",
+                                                  2 * projectedBaseHorizon / 2);
+    m_pimpl->structuredOutput.handler.addVariable("future_facing_directions",
+                                                  2 * projectedBaseHorizon / 2);
+    m_pimpl->structuredOutput.handler.addVariable("future_base_velocities",
+                                                  2 * projectedBaseHorizon / 2);
+    m_pimpl->structuredOutput.handler.addVariable("joint_positions", numberOfJoints);
+    m_pimpl->structuredOutput.handler.addVariable("joint_velocities", numberOfJoints);
+    m_pimpl->structuredOutput.handler.addVariable("base_velocity", 3);
+
+    // resize the output
+    m_pimpl->output.futureBasePositionTrajectory.resize(2, projectedBaseHorizon / 2);
+    m_pimpl->output.futureFacingDirectionTrajectory.resize(2, projectedBaseHorizon / 2);
+    m_pimpl->output.futureBaseVelocitiesTrajectory.resize(2, projectedBaseHorizon / 2);
+    m_pimpl->output.jointPositions.resize(numberOfJoints);
+    m_pimpl->output.jointVelocities.resize(numberOfJoints);
+    m_pimpl->output.projectedBaseVelocity = manif::SE2Tangentd::Zero();
+
+    return true;
+}
+
+bool MANN::setInput(const MANNInput& input)
+{
+    return m_pimpl->populateInput(input);
```

Comment:
Maybe it is worth checking that ``initialize`` has been called first.

Reply:
Done in https://github.com/ami-iit/bipedal-locomotion-framework/pull/652/commits/31b8e91aa7116c6b5642bd93c81420bd69fc8cb6